src/libs/antares/study/parts/thermal/cluster_list.cpp

```diff
@@ -159,51 +159,69 @@ bool ThermalClusterList::loadFromFolder(Study& study, const AnyString& folder, A
             options = Matrix<>::optFixedSize,
         };
 
-        bool r = cluster->modulation.loadFromCSVFile(modulationFile,
-                                                     thermalModulationMax,
-                                                     HOURS_PER_YEAR,
-                                                     options);
-        if (!r && study.usedByTheSolver)
+        ret = cluster->modulation.loadFromCSVFile(modulationFile,
+                                                  thermalModulationMax,
+                                                  HOURS_PER_YEAR,
+                                                  options) && ret;
+
+        // Check the data integrity of the cluster
+        addToCompleteList(cluster);
+    }
+
+    rebuildIndexes();
+    rebuildIndex();
+
+    return ret;
+}
+
+
+bool ThermalClusterList::validateClusters(const Parameters& parameters) const
+{
+    bool ret = true;
+
+    for (const auto& cluster : allClusters_)
+    {
+        if (cluster->minUpTime < 1)
```

Comment:
Please replace with: std::clamp(cluster->minUpTime, 1, 168)
And replace 1 and 168 with named constant if possible.

Reply:
It seems that some value "underflow". Some value stored as unint fall under 0, going back to UINT_MAX. When clamped they are clamped to 168.
At some point somewhere else we wrongly convert some value or don't check bounds. But it is outside of this PR to solve this.